components/compositing/compositor.rs

```diff
@@ -2002,24 +2261,43 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
         self.assert_gl_framebuffer_complete();
 
         // Set the viewport background based on prefs.
-        let viewport = self.embedder_coordinates.get_flipped_viewport();
-        gl.scissor(
-            viewport.origin.x,
-            viewport.origin.y,
-            viewport.size.width,
-            viewport.size.height,
-        );
-
         let color = servo_config::pref!(shell.background_color.rgba);
         gl.clear_color(
             color[0] as f32,
             color[1] as f32,
             color[2] as f32,
             color[3] as f32,
         );
-        gl.enable(gleam::gl::SCISSOR_TEST);
-        gl.clear(gleam::gl::COLOR_BUFFER_BIT);
-        gl.disable(gleam::gl::SCISSOR_TEST);
+
+        if cfg!(feature = "multiview") {
+            // Clear the viewport rect of each top-level browsing context.
+            for (_, webview) in self.webviews.painting_order() {
+                let rect = self
+                    .embedder_coordinates
+                    .flipped_rect(&webview.rect.to_i32());
+                gl.scissor(
+                    rect.origin.x,
+                    rect.origin.y,
+                    rect.size.width,
+                    rect.size.height,
+                );
+                gl.enable(gleam::gl::SCISSOR_TEST);
+                gl.clear(gleam::gl::COLOR_BUFFER_BIT);
+                gl.disable(gleam::gl::SCISSOR_TEST);
+            }
+        } else {
+            let viewport = self.embedder_coordinates.get_flipped_viewport();
+            gl.scissor(
+                viewport.origin.x,
+                viewport.origin.y,
+                viewport.size.width,
+                viewport.size.height,
+            );
+            gl.enable(gleam::gl::SCISSOR_TEST);
+            gl.clear(gleam::gl::COLOR_BUFFER_BIT);
+            gl.disable(gleam::gl::SCISSOR_TEST);
+        }
```

Comment:
Hrm... So all of these views use the same GL context? I think that's an issue. They should be isolated in GPU memory from one another.

Reply:
After considering this for a while, I think this isn't as large of an issue as I thought. Each page is sending display lists to WebRender which could potentially be in a different process along with this compositor. This separation should be enough.

Reply:
After considering this for a while, I think this isn't as large of an issue as I thought. Each page is sending display lists to WebRender which could potentially be in a different process along with this compositor. This separation should be enough.

Reply:
ok, no worries!